modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultConstantFoldingOptimizationPhase.java

```diff
@@ -1103,19 +1109,33 @@ private void replaceCallWithConstant(
         Object receiver
     ) {
         Object[] args = new Object[irInvokeCallMemberNode.getArgumentNodes().size()];
+
+        List<Tuple<Integer,String>> argumentsShouldBeConstants  = new ArrayList<>();
         for (int i = 0; i < irInvokeCallMemberNode.getArgumentNodes().size(); i++) {
             ExpressionNode argNode = irInvokeCallMemberNode.getArgumentNodes().get(i);
             IRDConstant constantDecoration = argNode.getDecoration(IRDConstant.class);
             if (constantDecoration == null) {
-                // TODO find a better string to output
-                throw irInvokeCallMemberNode.getLocation()
-                    .createError(
-                        new IllegalArgumentException(
-                            "all arguments to [" + javaMethod.getName() + "] must be constant but the [" + (i + 1) + "] argument isn't"
-                        )
-                    );
+                // offering the symbol name in error message (CastNode was evolved from ESymbol)
+                String argumentName = argNode instanceof CastNode
+                    ? ((CastNode) argNode).getChildNode().getDecoration(IRDName.class).getValue()
```

Comment:
If we go with this code (which is not clear at the moment), `getDecorationValue` should be used.

Reply:
OK, I will use `getDecorationValue`.